File: examples/ruby/history.rb
# Get candle history example
require_relative 'binary_options_tools_uni'

client = BinaryOptionsToolsUni::PocketOption.new("your-session-id")
sleep 5

# Get candle data for an asset
candles = client.get_candles("EURUSD_otc", 60, 3600)
puts "Raw Candles: #{candles}"

# Format and display individual candle data
candles.each do |candle|
  time = Time.at(candle["time"]).to_s
  puts "Time: #{time}, Open: #{candle["open"]}, High: #{candle["high"]}, Low: #{candle["low"]}, Close: #{candle["close"]}, Volume: #{candle["volume"]}"
end

File: examples/javascript/history.js
// Fetches candle history and formats the timestamps.
//
//   node history.js "<ssid>"

const { PocketOption } = require("../../nodejs");

async function main(ssid) {
  const api = await PocketOption.create(ssid);

  const candles = await api.history("EURUSD_otc", 3600);
  console.log(`Raw candles: ${candles.length}`);

  const formatted = candles.map((candle) => ({
    time: new Date(candle.time * 1000).toISOString(),
    open: candle.open,
    high: candle.high,
    low: candle.low,
    close: candle.close,
  }));

  console.log("Formatted candles:", formatted.slice(0, 5));

  await api.shutdown();
}

const ssid = process.argv[2] || process.env.POCKET_OPTION_SSID;
main(ssid).catch(console.error);
